sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -534,6 +501,9 @@ namespace Azure { namespace Core { namespace Http {
      * @brief Get URL.
      */
     Url const& GetUrl() const { return this->m_url; }
+    // Expected to be called by a Retry policy to reset all headers set after this function was
+    // previously called
+    void StartTry();
```

Comment:
What  does `StartTry` mean? This should be renamed.

Reply:
It is called at the beginning of each try & retry. DO you have a suggestion for a better name?

Reply:
Maybe `StartAttempt`. I will have to use the APIs and see it in sample code, in context, to give some suggestions.

What do other productized static languages do for this concept?